src/main.rs

```diff
@@ -21,45 +20,47 @@ const OTHER_BACKEND_NAME: &str = "other_backend_name";
 ///
 /// If `main` returns an error, a 500 error response will be delivered to the client.
 #[fastly::main]
-fn main(mut req: Request<Body>) -> Result<impl ResponseExt, Error> {
+fn main(mut req: Request) -> Result<Response, Error> {
     // Make any desired changes to the client request.
-    req.headers_mut()
-        .insert("Host", HeaderValue::from_static("example.com"));
+    req.set_header(header::HOST, "example.com");
 
-    // We can filter requests that have unexpected methods.
-    const VALID_METHODS: [Method; 3] = [Method::HEAD, Method::GET, Method::POST];
-    if !(VALID_METHODS.contains(req.method())) {
-        return Ok(Response::builder()
-            .status(StatusCode::METHOD_NOT_ALLOWED)
-            .body(Body::from("This method is not allowed"))?);
-    }
+    // Filter request methods...
+    match req.get_method() {
+        &Method::GET | &Method::HEAD => (),
```

Comment:
I would suggest a comment here for consistency with the other arms.  Otherwise at first glance it seems like we're denying all except PURGE requests.

Reply:
Fixed in 3be28ae! :sparkles: